transports/sql/design.md

```diff
@@ -1,78 +1,80 @@
-	[Id] [uniqueidentifier] NOT NULL,
-	[CorrelationId] [varchar](255),
-	[ReplyToAddress] [varchar](255),
-	[Recoverable] [bit] NOT NULL,
-	[Expires] [datetime],
-	[Headers] [varchar](max) NOT NULL,
-	[Body] [varbinary](max),
-	[RowVersion] [bigint] IDENTITY(1,1) NOT NULL
-)
-ON [PRIMARY];
 
-CREATE CLUSTERED INDEX [Index_RowVersion]
-ON [schema].[queuename]
-(
-	[RowVersion] ASC
-)
+### Recoverable
 
-CREATE NONCLUSTERED INDEX [Index_Expires]
-ON [schema].[queuename]
-(
-	[Expires] ASC
-)
-INCLUDE
-(
-	[Id],
-	[RowVersion]
-)
-```
+The `Recoverable` column should always contain the value `1` to ensure wire-level compatibility with transport Version 2 and lower.
 
-Receiving messages is conducted by a `DELETE` statement from the top of the table (the oldest row according to the `[RowVersion]` column).
 
-The tables are created by [installers](/nservicebus/operations/installers.md) when the application is started for the first time. It is required that the user account under which the installation of the host is performed has `CREATE TABLE` as well as `VIEW DEFINITION` permissions on the database in which the queues are to be created. The account under which the service runs does not have to have these permissions. Standard read/write/delete permissions (e.g. being member of `db_datawriter` and `db_datareader` roles) are enough.
+### Expires
 
+The `Expires` column contains the optional date and time when the message is going to expire. An expired message is dropped by the transport. Depending on version, expired messages might be actively purged from the queue. For details see [discarding expired messages](/transports/sql/discard-expired-messages.md).
 
-### Creating table structure in Production
+partial: expired-index
 
-The scripts above, to generate the queues, do not have queue names. This may cause confusion when reviewed by a DBA. The scripts could, alternatively, be generated off the Development or Staging environment, and then directly executed on Production environment by DBAs to replicate that table structure. 
+### Headers
 
-To generate this DDL script, right-click the database and from "Tasks" menu choose "Generate Scripts..." and generate the scripts for relevant tables.
+The `Headers` column contains a JSON representation of message headers.
 
-![](generating-ddl.png)
 
-partial: indexes
+### Body
+
+The `Body` column contains the serialized message body.
+
+
+### RowVersion
+
+The `RowVersion` column is used to define the FIFO order of the queue. It is auto-incremented by SQL Server (`identity(1,1)`). The receive message T-SQL query returns a message with the lowest value of `RowVersion` that is not locked by any other concurrent receive operation.
+
+The clustered index of the queue table is based on the `RowVersion` column to ensure the new messages are always added at the end of the table.
+
+
+## Behavior
+
+
+### Sending
+
+Messages are sent by executing an `insert` command against the queue table.
+
+
+### Receiving
```

Comment:
Should we add note that under high load the locks might get escalated from row-level and in such cases the interference between receivers will be bigger?

Reply:
Do we have proofs of this happening?